domain/application/state/metadata.go

```diff
@@ -217,7 +217,7 @@ func decodeExtraBindings(bindings []charmExtraBinding) map[string]charm.ExtraBin
 
 	result := make(map[string]charm.ExtraBinding)
 	for _, binding := range bindings {
-		result[binding.Key] = charm.ExtraBinding{
+		result[binding.Name] = charm.ExtraBinding{
 			Name: binding.Name,
 		}
```

Comment:
OT: I don't know why the extra binding ended up being a `map[string]string` when a `[]string` would have sufficed. It seems overly complicated.

Reply:
Best guess they did it to ensure that extra bindings were unique? Who knows at this point. Same for relations, why have a key and a name?